pkg/debugapi/balances.go

```diff
@@ -0,0 +1,72 @@
+package debugapi
+
+import (
+	"sort"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+
+	"net/http"
+)
+
+type balanceResponse struct {
+	Peer    string `json:"peer"`
+	Balance int64  `json:"balance"`
+}
+
+type balancesResponse struct {
+	Balances []balanceResponse `json:"balances"`
+}
+
+func (s *server) balancesHandler(w http.ResponseWriter, r *http.Request) {
+
+	balances, err := s.Accounting.Balances()
+
+	if err != nil {
+		// TODO: Do we need to return an explicit error?
+		s.Logger.Debugf("debug api: balances: %v", err)
```

Comment:
i would call internal server error on `w` here and `return`

Reply:
Something like
```go
jsonhttp.InternalServerError(w, "cannot get balances")
```
should do the trick. That's how the other packages seem to do it.

Reply:
but, even if we have an error from Balances(), we probably got most of it right from memory (if statestore has some problem)
I'd like to keep this as is

Reply:
the problem with this approach is that if you ignore `err` on `balancesHandler` (`balances.go:28`) and return a partial resort, you're still doing a `jsonhttp.OK` at the end of the call.

how could the caller tell apart a response with an error from one without one?

e.g.:
- empty `Balances` response: did something go wrong, or are there just no balances?
- non-empty `Balances` response: is this the full list of balances, or am i missing some because an `err` occurred somewhere along the way?

if you keep as is, you need to supply some sort of error info along with the result through `jsonhttp`

Reply:
@mortelli We were considering this approach, however, does the client care if the internal store access is compromised in this case?

Also, if the internal store is not accessible/responding - what should we return as the result code?

Reply:
guys, if there's an error from statestore make the api return an explicit error message without a balance response. otherwise we end up with a whole lot of messy code on the clients that have to implement these apis. also, statestore error is very severe, and suggests a whole lot of other inconsistency errors which are happening. we should assume the node is already misbehaving/malfunctioning and treat it as such, and not consider the response as `200 OK` in these kind of scenarios